lib/graphite-api/buffer.rb

```diff
@@ -62,29 +74,35 @@ def push obj
     alias_method :<<, :push
 
     def pull format = nil
-      data = Hash.new {|h,k| h[k] = Hash.new {|h2,k2| h2[k2] = 0}}
+      data = Hash.new { |h,time| h[time] = Hash.new { |h2,metric| h2[metric] = cache ? cache.get(time, metric) : nil } }
+      aggregation_methods = Hash.new { |h, time| h[time] = options[:aggregation_method] }
```

Comment:
Instead of building and using this Hash object, can you do it inline ?

```ruby
# lines 86-87
 data[normalized_time][metric] = AGGREGATORS[method || options[:aggregation_method]].call(data[normalized_time][metric], value.to_f)
```